server/chat-plugins/datasearch.ts

```diff
@@ -10,15 +10,53 @@
 
 'use strict';
 
+import {QueryProcessManager} from '../../lib/process-manager';
+
+interface DexOrGroup {
+	abilities: {[k: string]: boolean};
+	tiers: {[k: string]: boolean};
+	doublesTiers: {[k: string]: boolean};
+	colors: {[k: string]: boolean};
+	'egg groups': {[k: string]: boolean};
+	formes: {[k: string]: boolean};
+	gens: {[k: string]: boolean};
+	moves: {[k: string]: boolean};
+	types: {[k: string]: boolean};
+	resists: {[k: string]: boolean};
+	weak: {[k: string]: boolean};
+	stats: {[k: string]: {[k in Direction]: number}};
+	skip: boolean;
+}
+
+interface MoveOrGroup {
+	types: {[k: string]: boolean};
+	categories: {[k: string]: boolean};
+	contestTypes: {[k: string]: boolean};
+	flags: {[k: string]: boolean};
+	gens: {[k: string]: boolean};
+	recovery: {[k: string]: boolean};
+	mon: {[k: string]: boolean};
+	property: {[k: string]: {[k in Direction]: number}};
+	boost: {[k: string]: boolean};
+	lower: {[k: string]: boolean};
+	zboost: {[k: string]: boolean};
+	status: {[k: string]: boolean};
+	volatileStatus: {[k: string]: boolean};
+	recoil: boolean;
+	skip: boolean;
+}
+
+type Direction = 'less' | 'greater' | 'equal';
```

Comment:
I'm not _particularly_ happy with this either, but I guess it's used for some minor safety in some places, we can keep it.

Reply:
I realised its not what you're complaining about, but I wonder why we went with the english words instead of the more concise `'<' | '=' | '>'` instead